calicoctl/commands/resources.go

```diff
@@ -41,15 +38,15 @@ const (
 	actionCreate
 	actionUpdate
 	actionDelete
-	actionList
+	actionGetOrList
 )
 
 // Convert loaded resources to a slice of resources for easier processing.
 // The loaded resources may be a slice containing resources and resource lists, or
 // may be a single resource or a single resource list.  This function handles the
 // different possible options to convert to a single slice of resources.
-func convertToSliceOfResources(loaded interface{}) []unversioned.Resource {
-	r := []unversioned.Resource{}
+func convertToSliceOfResources(loaded interface{}) []runtime.Object {
```

Comment:
This is a actually converting to a slice of ResourceObjects, so perhaps that's what this should return.  I think it'll make the rest of the plumbing much simpler

Reply:
Remember that a runtime.Object really is just something with a Kind/Version so that covers both Lists and Objects - but in this particular case we should only be returning Objects.  Probably need to beef up the function comment to make that clearer.